### PR TITLE
support controling visiblity of system tray menu items

### LINKED ISSUE #3095: Support to control visibility of system tray MenuItems
### Is your feature request related to a problem? Please describe:
I would like to be able to control the visibility of a systrayMenu `MenuItem`.

### Is it possible to construct a solution with the existing API?C
Currently we have to add/remove items from the menu and then refresh it.

### Describe the solution you'd like to see:
Add a `Visible` attribute to `MenuItem`, evalute it in `refreshSystray` and call `Hide()` or `Show()` on the underlying `systray.MenuItem` accordingly.

### PR DESCRIPTION
**Pull request by d1ss0nanz**
### Description:
Add a `Visible` attribute to `MenuItem`, evalute it in `refreshSystray` and call `Hide()` or `Show()` on the underlying `systray.MenuItem` accordingly.

Fixes #3095

### Checklist:

- [ ] Tests included.
- [x] Lint and formatter run with no errors.
- [ ] Tests all pass.

#### Where applicable:

- [x] Public APIs match existing style and have Since: line.

### CODE REVIEW

**Comment by d1ss0nanz** [comment]
Good points.

Will work on it.

**Review by Bluebugs — COMMENTED** [review]
I just wonder if it would be possible to have some test for this. Otherwise it looks pretty good.

File: menu.go [hunk 1]
```diff
@@ -57,6 +57,8 @@ type MenuItem struct {
 	Icon Resource
 	// Since: 2.2
 	Shortcut Shortcut
+	// Since: 2.2.2
```
Comment (andydotxyz) [inline menu.go:60]: We don't allow API changes in patch releases so this will need to be for 2.3.
This is also why we only use 2 digits for API level :)

**Review by andydotxyz — CHANGES_REQUESTED** [review]
Looks good, but the API version is a bit too keen!

**Comment by andydotxyz** [comment]
> I just wonder if it would be possible to have some test for this. Otherwise it looks pretty good.

This might be tricky as I don't think we have test infrastructure in place for system tray Fyne integration.
That said we could literally add a system tray icon and run the code - though it is starting to get intrusive for development. Perhaps it can be pushed back and added with other "full-test" flag that has been discussed before?
(Allowing all unit tests other than this tag to run without showing anything on screen...)

File: menu.go (hunk 1 above)
Comment (d1ss0nanz) [reply menu.go:60]: Done.

**Comment by d1ss0nanz** [comment]
> I just wonder if it would be possible to have some test for this. Otherwise it looks pretty good.
Sure. Where should I put the tests?
There must be a bunch of related tests, where I just can extend the coverage, right?

**Comment by andydotxyz** [comment]
I think your last commit was wrong @d1ss0nanz - it added a blank file to a `cmd/systray_demo` package. Let's not have more executables. The system tray is demoed in fyne_demo like everything else.

As for unit testing I expect that alongside the menu_test in drivers would be sufficient. We can't really test the systray aspects as that rendering is inside the `fyne.io/systray` package.

**Comment by Jacalz** [comment]
Hi @d1ss0nanz. Do you plan on continuing this PR? It is currently out of date from develop so you will need to rebase on develop to get to pass tests. I believe the last message from Andrew should contain solutions to the remaining work before we can get this merged :)

**Comment by andydotxyz** [comment]
It's been almost a year since last commit and no response in 2 months - closing.

**Pull request closed without merging**